From: Richard Hansen <rhansen@rhansen.org>
Date: Fri, 6 Mar 2026 02:00:08 -0500
Subject: estimate: Delete unnecessary removeVendor

Downloaded Go modules do not have any `vendor` directories since Go
ignores them when creating a zip archive for the
GOPROXY (proxy.golang.org) service:
<https://go.dev/ref/mod#zip-files>.  Thus, there's no value in
searching for them.
---
 estimate.go | 32 --------------------------------
 1 file changed, 32 deletions(-)

diff --git a/estimate.go b/estimate.go
index 40bb4ac..f90c11a 100644
--- a/estimate.go
+++ b/estimate.go
@@ -143,26 +143,6 @@ func getDirectDependencies(gopath, repodir, module string) (map[string]bool, err
 	return deps, nil
 }
 
-func removeVendor(repodir string) (found bool, _ error) {
-	err := filepath.Walk(repodir, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if !info.IsDir() {
-			return nil // skip non-directories
-		}
-		if info.Name() != "vendor" {
-			return nil
-		}
-		found = true
-		if err := os.RemoveAll(path); err != nil {
-			return fmt.Errorf("remove all: %w", err)
-		}
-		return filepath.SkipDir
-	})
-	return found, err
-}
-
 // otherVersions guesses the import paths of potential other major version
 // of the given module import path, based on [majorVersionRegex].
 func otherVersions(mod string) (mods []string) {
@@ -237,18 +217,6 @@ func estimate(importpath, revision string) error {
 		return fmt.Errorf("go get: %w", err)
 	}
 
-	found, err := removeVendor(repodir)
-	if err != nil {
-		return fmt.Errorf("remove vendor: %w", err)
-	}
-
-	if found {
-		// Fetch un-vendored dependencies
-		if err := get(gopath, repodir, importpath, revision); err != nil {
-			return fmt.Errorf("fetch un-vendored: go get: %w", err)
-		}
-	}
-
 	// Get dependency graph from go mod graph
 	cmd := exec.Command("go", "mod", "graph")
 	cmd.Dir = repodir
